lib/datadog/tracing/configuration/settings.rb

```diff
@@ -175,6 +176,44 @@ def self.extended(base)
                 o.lazy
               end
 
+              # Enable 128 bit trace id generation.
+              #
+              # @default `DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED` environment variable, otherwise `false`
+              # @return [Boolean]
+              option :trace_id_128_bit_generation_enabled do |o|
+                o.default { env_to_bool(Tracing::Configuration::Ext::ENV_TRACE_ID_128_BIT_GENERATION_ENABLED, false) }
+                o.lazy
+              end
+
+              # Enable 128 bit trace id propagation for distributed tracing.
+              #
+              # @default `DD_TRACE_128_BIT_TRACEID_PROPAGATION_ENABLED` environment variable, otherwise `false`
+              # @return [Boolean]
+              option :trace_id_128_bit_propagation_enabled do |o|
+                o.default { env_to_bool(Tracing::Configuration::Ext::Distributed::ENV_TRACE_ID_128_BIT_PROPAGATION_ENABLED, false) }
+                o.lazy
+                o.depends_on [:trace_id_128_bit_generation_enabled]
+
+                o.on_set do |v|
+                  if get_option(:trace_id_128_bit_generation_enabled) && v != true
+                    Datadog.logger.warn(
+                      '128 bit trace_id is generated but not propagated. '\
+                      "Enable environment variable `#{Tracing::Configuration::Ext::Distributed::ENV_TRACE_ID_128_BIT_PROPAGATION_ENABLED}` with `true` "\
+                      "to enable option `#{o.instance_variable_get(:@name)}`"
+                    )
+                  end
+                end
+              end
+
+              # Enable 128 bit trace id injected for logging.
+              #
+              # @default `DD_TRACE_128_BIT_TRACEID_LOGGING_ENABLED` environment variable, otherwise `false`
+              # @return [Boolean]
+              option :trace_id_128_bit_logging_enabled do |o|
```

Comment:
Should we also emit a warning here when 128-bit is enabled but this is not?

Reply:
I think that make sense!